### PR TITLE
Added support to insert a reference to an organization or a role into operatorRef.

### PR DESCRIPTION
**Pull request by mokracekjan**
Members of the organizations (direct only) or roles can be used as operators of manual applications - only org:default relation.
- It enables more robust and automated management of operators of individual manual resources.

### CODE REVIEW

**Comment by martin-lizner** [comment]
Nice, thank you :-)

**Comment by mederly** [comment]
Nice! I would suggest specifying also the relation in the query, to avoid collecting e.g. approvers or owners. Actually, it is not quite trivial to do so, because of the custom relations that might be configured for a specific deployment. You could either construct a query selecting only member-like relations (looks like {{relationRegistry.getAllRelationsFor(MEMBER)}} could be used for this; but then the query construction is more complex because of multiple relations). Or you can get all the members, and filter-out non-members before the use.

**Comment by mokracekjan** [comment]
Thank you for your proposal, **but this is how it is already implemented**. When a query includes a reference that was created using a constructor that takes only the target's oid as a parameter, only the objects that have the target object assigned in the **org: default** relation (members only) are searched when using that query. I also performed the following test:
- I assigned an operator role and organization to 5 users in all possible relations (I even used a **custom relation**), work items were actually created only for users who had a given operator role or organization assigned in a member relation.
- In other words, **all other relation types**, except org: default, **have been filtered out**.

**Comment by mederly** [comment]
That's interesting. :) I have seen the code and I was sure the constructed filter should match all values of the relation. I will have a look at the current implementation of the filter.

**Comment by mokracekjan** [comment]
I also thought it worked like this, but some time ago (on the customer's environment) I found out that only **org: default** is searched without specifying the relation type.

**Comment by virgo47** [comment]
Interpreting non-specified relation as default *for reference filter* is indeed in line with our documentation: https://docs.evolveum.com/midpoint/reference/concepts/query/query-api/
Search for "default" word and it will say exactly this in a summary table and then in explicit explanation how relation is processed in ref filter (which is the case of this PR if I understand it correctly) and Org filter (which is not used here). In Org filter it works as "any" relation.

**Comment by mederly** [comment]
@virgo47 Thanks. Then it's all right. I was just confused.

**Comment by mokracekjan** [comment]
Thank you, therefore the query in the contribution should be correct.

**Comment by mederly** [comment]
@mokracekjan Thank you for the contribution. I have merged it just now.

I would have a little question for you: When adding a new feature, we create also a test for it. (Typically by adding a test method to existing test class, sometimes even by extending existing test method.) Would you try to write such a test?

Here is general documentation related to writing tests in midPoint: https://docs.evolveum.com/midpoint/devel/testing/integration/. I have looked for suitable place for a test for this feature. It could be be either in `provisioning-impl` module (under [AbstractManualResourceTest](https://github.com/Evolveum/midpoint/blob/master/provisioning/provisioning-impl/src/test/java/com/evolveum/midpoint/provisioning/impl/manual/AbstractManualResourceTest.java) or in `model-intest` module (under [AbstractManualResourceTest](https://github.com/Evolveum/midpoint/blob/802fbb459e733ba384566502402658abbaffae2b/model/model-intest/src/test/java/com/evolveum/midpoint/model/intest/manual/AbstractManualResourceTest.java)).

**Comment by mokracekjan** [comment]
@mederly Sure, i 'll try that.
